### PR TITLE
Make namespace controller delete configMaps in namespace

### PR DESCRIPTION
**Pull request by pmorie**
@derekwaynecarr

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
In this PR (https://github.com/kubernetes/kubernetes/pull/20076), I am trying to move all calls that can support it to use `DeleteCollection`, can you make this use that call instead?

File: pkg/controller/namespace/namespace_controller.go [hunk 1]
```diff
@@ -454,6 +458,20 @@ func deleteSecrets(kubeClient clientset.Interface, ns string) error {
 	return nil
 }
 
+func deleteConfigMaps(kubeClient clientset.Interface, ns string) error {
+	items, err := kubeClient.Legacy().ConfigMaps(ns).List(api.ListOptions{})
```
Comment (derekwaynecarr) [inline pkg/controller/namespace/namespace_controller.go]: DeleteCollection()

**Comment by pmorie** [comment]
@k8s-bot test this issue: #20694

**Comment by pmorie** [comment]
@k8s-bot test this issue: #20916

**Comment by derekwaynecarr** [comment]
Can we merge https://github.com/kubernetes/kubernetes/pull/20076 first?

**Comment by pmorie** [comment]
@derekwaynecarr my bad, I had forgotten about the dependency on your pull.

**Comment by derekwaynecarr** [comment]
rebase then i will lgtm

**Pull request merged**